### PR TITLE
Do not rely on enum values in c++ conversion code

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: dataplane/apigen/docparser/docparser.go [hunk 1]
```diff
@@ -112,25 +113,44 @@ func ParseSAIXMLDir() (*SAIInfo, error) {
 	if err != nil {
 		return nil, err
 	}
+	members := []MemberDef{}
 	for _, file := range files {
 		if !strings.HasSuffix(file.Name(), "xml") {
 			continue
 		}
-		if err := parseXMLFile(filepath.Join(xmlPath, file.Name()), i); err != nil {
+		if strings.Contains(file.Name(), "saimetadata") {
```
Comment (guoshiuan) [inline dataplane/apigen/docparser/docparser.go:121]: Nit: probably make a func that checks the filename to skip files?

File: dataplane/apigen/docparser/docparser.go [hunk 2]
```diff
@@ -171,62 +196,138 @@ func memberToAttrInfo(enum MemberDef) *Attr {
...
+	init := strings.TrimSpace(strings.TrimPrefix(vals[i].Initializer, "= "))
+	if init == "" {
+		if i == 0 {
+			return 0, nil
+		}
+		prev, err := parseInitializer(i-1, vals)
+		return prev + 1, err
+	}
+	if strings.Contains(init, "+") {
+		splits := strings.Split(init, "+")
```
Comment (guoshiuan) [inline dataplane/apigen/docparser/docparser.go:231]: Do we want to assert splits has exactly 2 elements?

File: dataplane/apigen/docparser/docparser.go [hunk 3]
```diff
@@ -171,62 +196,138 @@ func memberToAttrInfo(enum MemberDef) *Attr {
...
+			rv = uint64(saiConst)
+		}
+		lv, err := parseInitializer(findIndex(lhs, vals), vals)
+		if err != nil {
+			return 0, err
+		}
+		return lv + int(rv), nil
+	}
+	if strings.Contains(init, "<<") {
+		splits := strings.Split(init, "<<")
```
Comment (guoshiuan) [inline dataplane/apigen/docparser/docparser.go:249]: same as above.

File: dataplane/apigen/docparser/docparser.go [hunk 4]
```diff
@@ -171,62 +196,138 @@ func memberToAttrInfo(enum MemberDef) *Attr {
...
+	}
+
+	for _, enum := range members {
+		err := processEntry(enum)
+		if err != nil {
+			retryEntry = append(retryEntry, enum)
+		}
+	}
+	consts := saiConsts
+	fmt.Println(consts)
```
Comment (guoshiuan) [inline dataplane/apigen/docparser/docparser.go:325]: Delete this println?

**Review by guoshiuan — APPROVED** [review]

File: dataplane/apigen/docparser/docparser.go (hunk 4 above)
Comment (DanG100) [reply dataplane/apigen/docparser/docparser.go]: done

File: dataplane/apigen/docparser/docparser.go (hunk 2 above)
Comment (DanG100) [reply dataplane/apigen/docparser/docparser.go:231]: done

File: dataplane/apigen/docparser/docparser.go (hunk 3 above)
Comment (DanG100) [reply dataplane/apigen/docparser/docparser.go:252]: done

File: dataplane/apigen/docparser/docparser.go (hunk 1 above)
Comment (DanG100) [reply dataplane/apigen/docparser/docparser.go:121]: will do if this needs another case

**Pull request merged**